scalyr_agent/configuration.py

```diff
@@ -223,6 +223,58 @@ def parse(self):
                 if https_server != server:
                     self.__config["scalyr_server"] = https_server
 
+            # Set defaults based on `max_send_rate_enforcement` value
+            if not self.__config["disable_max_send_rate_enforcement_overrides"]:
+                self._warn_of_override_due_to_rate_enforcement(
+                    "max_allowed_request_size", 1024 * 1024
+                )
+                self._warn_of_override_due_to_rate_enforcement(
+                    "pipeline_threshold", 1.1
+                )
+                self._warn_of_override_due_to_rate_enforcement(
+                    "min_request_spacing_interval", 1.0
+                )
+                self._warn_of_override_due_to_rate_enforcement(
+                    "max_request_spacing_interval", 5.0
+                )
+                self._warn_of_override_due_to_rate_enforcement(
+                    "max_log_offset_size", 5 * 1024 * 1024
+                )
+                self._warn_of_override_due_to_rate_enforcement(
+                    "max_existing_log_offset_size", 100 * 1024 * 1024
+                )
+
+                if self.__config["max_send_rate_enforcement"] == "legacy":
+                    self.__config["max_allowed_request_size"] = 1024 * 1024
```

Comment:
Ok... what happens here if the customer has set a value for `max_allowed_request_size`?  If `legacy` is selected, then we should either use the old default value or the value from the configuration file if it has been set.

Reply:
Currently it forces the old default value (I should add a test for that too actually).

Reply:
Right, but we want it do is:  if `legacy` is selected, first use the value from the configuration file, or if there is not a value, the old default value.